scripts/migrate_users_to_keycloak.pl

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/perl -w
+
+# This file is part of Product Opener.
+#
+# Product Opener
+# Copyright (C) 2011-2023 Association Open Food Facts
+# Contact: contact@openfoodfacts.org
+# Address: 21 rue des Iles, 94100 Saint-Maur des Fossés, France
+#
+# Product Opener is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use ProductOpener::PerlStandards;
+
+use ProductOpener::Auth qw/:all/;
+use ProductOpener::Config qw/:all/;
+use ProductOpener::Paths qw/:all/;
+use ProductOpener::Store qw/:all/;
+
+use JSON;
+use LWP::UserAgent;
+use LWP::UserAgent::Plugin 'Retry';
+use HTTP::Request;
+use URI::Escape::XS qw/uri_escape/;
+
+sub create_user_in_keycloak_with_scrypt_credential ($user_ref, $credential) {
+	my $keycloak_users_endpoint = $oidc_options{keycloak_users_endpoint};
+	unless ($keycloak_users_endpoint) {
+		die 'keycloak_users_endpoint not configured';
+	}
+
+	my $token = get_token_using_client_credentials();
+	unless ($token) {
```

Comment:
tbd: Reuse this token until it's close to expiring. Getting a new token for each interaction works in dev, but won't if we want to migrate thousands of users.